File: TEO/Mav.Services/Abstract/IOrganisationService.cs
using Mav.Common.Models;
using Mav.Data.Repositories;
using Mav.Models.SearchModels;
using Mav.Models.ViewModels;
using System;
using System.Collections.Generic;
using System.Text;
using System.Threading.Tasks;

namespace Mav.Services.Abstract
{
    public interface IOrganisationService
    {
        Task<T> GetOrganisationById<T>(int id);

        Task<IPaginate<OrganisationListingModel>> GetAllOrganisations(OrganisationSearchModel vm);

        Task<GenericResult<T>> AddOrganisation<T>(AddOrganisationViewModel m);

        Task<GenericResult<T>> Update<T>(UpdateOrganisationViewModel m);

        Task<GenericResult<T>> Delete<T>(int id);
    }
}

File: TEO/Mav.UI/Mav.Ui/src/app/services/i-organisation.service.ts
import { Observable } from "rxjs/Observable";
import { OrganisationViewModel } from "../models/admin/organisation.view.model";
import { OrganisationSearchModel } from "../models/search/organisation.search.model";
import { OrganisationListingModel } from "../models/admin/organisation.listing.model";
import { Paginate } from "../models/paginated.items";
import { AddOrganisationViewModel } from "../models/admin/add-organisation.view.model";
import { UpdateOrganisationViewModel } from "../models/admin/update-organisation.view.model";
import { GenericResult } from "../models/generic.result.model";

export interface IOrganisationService {

    getOrganisationById(id: number): Observable<OrganisationViewModel>;

    getAllOrganisations(filter: OrganisationSearchModel): Observable<Paginate<OrganisationListingModel>>;

    createOrganisation(org: AddOrganisationViewModel): Observable<GenericResult<number>>;

    updateOrganisation(org: UpdateOrganisationViewModel): Observable<GenericResult<number>>;

    deleteOrganisation(id: number): Observable<GenericResult<number>>;
}